cloud/main.js

```diff
@@ -33,6 +33,65 @@ Parse.Cloud.define("sendMessage", function(request, response) {
                        });
 });
 
+Parse.Cloud.define("registerNewUser", function(request, response) {
+    getUserCount(function(userCount) {
+        if (userCount == undefined) {
+            response.error;
+        }
+
+        var user = new Parse.User();
+        user.set('password', 'password');
+        user.set('username', (userCount + 1).toString());
+
+        user.save(null, {
+            success: function(user) {
+                incrementUserCount(response, function() {
+                    response.success(user);
+                });
+            },
+            error: function(message, error) {
+                response.error(error);
+            }
+        });
+    });
+});
+
+function incrementUserCount(response, successCallback) {
+    var UserCount = Parse.Object.extend("UserCount");
+    var userCountQuery = new Parse.Query(UserCount);
+
+    userCountQuery.get("SqRF3QUGsM", {  
+        success: function(userCount) {
+            userCount.increment('count');
+            userCount.save(null, {
+                success: function(userCount) {
+                    successCallback();
+                },
+                error: function(object,error) {
+                    response.error;
+                }
+            });
+        },
+        error: function(object,error) {
+            response.error(error);
+        }
+    });
+}
+
+function getUserCount(callback) {
+    var UserCount = Parse.Object.extend("UserCount");
+    var userCountQuery = new Parse.Query(UserCount);
+
+    userCountQuery.get("SqRF3QUGsM", { 
+        success: function(userCount) {
+            callback(userCount.get("count"));
+        },
+        error: function(object,error) {
+            callback(undefined);
```

Comment:
Here, you have a single callback for success and error and the only difference is what the value is. I'd almost argue that it's better to take a {success,error} response object here and call the appropriate callback depending on the result. Reason being: callback(undefined) makes it clear that this function failed but doesn't actually report what the error was and makes debugging harder